### PR TITLE
Pipefail

### PR DESCRIPTION
**Pull request by afishman**
This PR resolves a unit testing bug whereby errors in mask.bash were being left uncaught. This was because the error was raised within a pipe on a bash script. To resolve this, and prevent similar issues arising in other nf processes - `pipefail` and `set -e` have been added to all nextflow bash scripts.

### CODE REVIEW

**Review by ellisrichardj — COMMENTED** [review]
I'd missed that these didn't have set -e already, so certainly great to have it added.  Is there any reason these additions aren't immediately after the shebang where they are usually found?

**Comment by ellisrichardj** [comment]
I actually think pipefail is classed as an option: https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html, so perhaps it should be `set -o pipefail`?  or combined to a single line: `set -eo pipefail`

**Comment by afishman** [comment]
great idea -- the `set -eo pipefail` has worked a treat. I've updated all the nf processes with that.

File: bin/assignClusterCss.bash [hunk 1]
```diff
@@ -14,6 +14,10 @@ min_qual_nonsnp=$8
 pypath=$9
 ref=$10
 
+# Error handling
+set -e
+set pipefail
```
Comment (ellisrichardj) [inline bin/assignClusterCss.bash:19]: Can you change this one to `set -eo pipefail` as well

**Review by ellisrichardj — COMMENTED** [review]
I wonder if the use of `#%` for comment lines rather than the bash standard `#`  may be causing the issue with loosing variables?

**Comment by afishman** [comment]
Done. The `#%` does not make a difference. Actually it's because the `set` command overwrites input arguments when `-` or `+` is not used to signify options! See:
https://phoenixnap.com/kb/linux-set#:~:text=The%20set%20command%20is%20a,and%20Korn%20shell%20(%20ksh%20).

**Comment by ellisrichardj** [comment]
Ahh, in which case, the problem was due to the initial use of `set pipefail` as its not proceeded by a `-`.  Having this specified as `set -eo pipefail` fixes that issue with some testing I have done

**Comment by afishman** [comment]
Yeah exactly. So are you happy to approve now?

**Review by ellisrichardj — APPROVED** [review]
Great  - all good now

**Pull request merged**